tick/stateful_expr_test.go

```diff
@@ -0,0 +1,398 @@
+
+		// Left: False, Right: "NON_BOOL_VALUE"
+		keyStruct{false, "NON_BOOL_VALUE", tick.TokenEqual}:    errors.New("mismatched type to binary operator. got bool == string. see bool(), int(), float()"),
+		keyStruct{false, "NON_BOOL_VALUE", tick.TokenNotEqual}: errors.New("mismatched type to binary operator. got bool != string. see bool(), int(), float()"),
+		keyStruct{false, "NON_BOOL_VALUE", tick.TokenAnd}:      errors.New("mismatched type to binary operator. got bool AND string. see bool(), int(), float()"),
+		keyStruct{false, "NON_BOOL_VALUE", tick.TokenOr}:       errors.New("mismatched type to binary operator. got bool OR string. see bool(), int(), float()"),
+
+		// Left: "NON_BOOL_VALUE", Right: True
+		keyStruct{"NON_BOOL_VALUE", true, tick.TokenEqual}:    errors.New("mismatched type to binary operator. got string == bool. see bool(), int(), float()"),
+		keyStruct{"NON_BOOL_VALUE", true, tick.TokenNotEqual}: errors.New("mismatched type to binary operator. got string != bool. see bool(), int(), float()"),
+		keyStruct{"NON_BOOL_VALUE", true, tick.TokenAnd}:      errors.New("mismatched type to binary operator. got string AND bool. see bool(), int(), float()"),
+		keyStruct{"NON_BOOL_VALUE", true, tick.TokenOr}:       errors.New("mismatched type to binary operator. got string OR bool. see bool(), int(), float()"),
+
+		// Left: "NON_BOOL_VALUE", Right: False
+		keyStruct{"NON_BOOL_VALUE", false, tick.TokenEqual}:    errors.New("mismatched type to binary operator. got string == bool. see bool(), int(), float()"),
+		keyStruct{"NON_BOOL_VALUE", false, tick.TokenNotEqual}: errors.New("mismatched type to binary operator. got string != bool. see bool(), int(), float()"),
+		keyStruct{"NON_BOOL_VALUE", false, tick.TokenAnd}:      errors.New("mismatched type to binary operator. got string AND bool. see bool(), int(), float()"),
+		keyStruct{"NON_BOOL_VALUE", false, tick.TokenOr}:       errors.New("mismatched type to binary operator. got string OR bool. see bool(), int(), float()"),
+	})
+
+}
+
+func TestStatefulExpression_EvalBool_NumberNode(t *testing.T) {
+	leftValues := []interface{}{float64(5), float64(10), int64(5)}
+	rightValues := []interface{}{float64(5), float64(10), int64(5), "NON_INT_VALUE"}
+
+	operators := []tick.TokenType{tick.TokenEqual, tick.TokenNotEqual, tick.TokenGreater, tick.TokenGreaterEqual, tick.TokenLessEqual, tick.TokenLess}
+
+	createNumberNode := func(v interface{}) tick.Node {
+		if reflect.TypeOf(v).Kind() == reflect.Int64 {
+			return &tick.NumberNode{
+				IsInt: true,
+				Int64: v.(int64),
+			}
+		}
+
+		if reflect.TypeOf(v).Kind() == reflect.Float64 {
+			return &tick.NumberNode{
+				IsFloat: true,
+				Float64: v.(float64),
+			}
+		}
+
+		// For the error case
+		if reflect.TypeOf(v).Kind() == reflect.String {
+			return &tick.StringNode{
+				Literal: v.(string),
+			}
+		}
```

Comment:
Instead of using reflect can we just use a type switch?

```go
switch value:=v.(type){
 case int64:
    return &tick.NumberNode{
				IsInt: true,
				Int64: value,
			}
case float64:
    return &tick.NumberNode{
 				IsFloat: true,
 				Float64: value,
 			}
//etc...
}
```